external/fv3fit/fv3fit/_shared/config.py

```diff
@@ -2,15 +2,128 @@
+T = TypeVar("T", bound="Loadable")
 
 DELP = "pressure_thickness_of_atmospheric_layer"
 MODEL_CONFIG_FILENAME = "training_config.yml"
 
 
+KERAS_MODELS: Dict[str, type] = {}
+SKLEARN_MODEL_TYPES = ["sklearn", "rf", "random_forest", "sklearn_random_forest"]
+
+
+def get_keras_model(name):
+    return KERAS_MODELS[name]
+
+
+def register_keras_trainer(name):
+    def decorator(cls):
+        KERAS_MODELS[name] = cls
+        return cls
+
+    return decorator
+
+
+class Dumpable:
+    def dump(self, filename: str) -> None:
+        dict_ = dataclasses.asdict(self)
+        with fsspec.open(filename, "w") as f:
+            yaml.safe_dump(dict_, f)
+
+
+class Loadable:
+    @classmethod
+    def load(cls: Type[T], filename: str) -> T:
+        with fsspec.open(filename, "r") as f:
+            config_dict = yaml.safe_load(f)
+        return cls(**config_dict)  # type: ignore
+
+
+@dataclasses.dataclass
+class DataConfig(Dumpable):
+    """Convenience wrapper for model training data
+
+    Attrs:
+        variables: names of variables to include in dataset
+        batch_function: name of function from `fv3fit.batches` to use for
+            loading batched data
+        batch_kwargs: keyword arguments to pass to batch function
+        data_path: location of training data to be loaded by batch function
+        validation_timesteps: timestamps to use as validation samples
+        timesteps_source: one of "timesteps_file",
+            "sampled_outside_input_config", "input_config", "all_mapper_times"
+    """
+
+    variables: List[str]
+    data_path: str
+    batch_function: str
+    batch_kwargs: dict
+
+
 @dataclasses.dataclass
-class ModelTrainingConfig:
+class TrainingConfig(Dumpable):
+    """Convenience wrapper for model training parameters and file info
+
+    Attrs:
+        model_type: sklearn model type or keras model class to initialize
+        hyperparameters: arguments to pass to model class at initialization
+            time
+        input_variables: variables used as features
+        output_variables: variables to predict
+        batch_function: name of function from `fv3fit.batches` to use for
+            loading batched data
+        batch_kwargs: keyword arguments to pass to batch function
+        data_path: location of training data to be loaded by batch function
+        scaler_type: scaler to use for training
+        scaler_kwargs: keyword arguments to pass to scaler initialization
+        additional_variables: list of needed variables which are not inputs
+            or outputs (e.g. pressure thickness if needed for scaling)
+        random_seed: value to use to initialize randomness
+        validation_timesteps: timestamps to use as validation samples
+        save_model_checkpoints: whether to save a copy of the model at
+            each epoch
+        model_path: output location for final model
+        timesteps_source: one of "timesteps_file",
```

Comment:
Remove extra args that are no longer in this dataclass

Reply:
Thanks, done. Added docstrings to the subclasses which include their additional options.